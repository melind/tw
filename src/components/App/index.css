.app {
  height: 100%;
}

.title {
  font-family: 'Modak', cursive;
  font-size: 400%;
  text-align: center;
  height: 5%;
  margin-top: 0;
}



