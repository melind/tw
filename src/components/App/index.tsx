import React, { useState } from 'react';
import { Route, Switch, Redirect} from 'react-router-dom';
import userAPI from '../../services/userAPI';
import './index.css';

import Home from '../Home';
import H from '../H';
import Signup from '../../containers/Signup';
import Login from '../../containers/Login';
import Account from '../../containers/Account';
import Pseudo from '../../containers/Pseudo';
import Mail from '../../containers/Mail';
import Password from '../../containers/Password';
import NotFoundPage from '../NotFoundPage';
import Admin from '../Admin';
import Load from '../Load';
import Forbidden from '../Forbidden';
import Media from '../Media';
import Genre from '../Genre';
import GenreTv from '../GenreTv';
import Mentions from '../Mentions';
import PrivateRoute from '../PrivateRoute';
import PublicRoute from '../PublicRoute';

const App = () => {
  userAPI.index();

  const [loading, setLoading] = useState(true);
    
     setTimeout(function () {
      setLoading(false);
    }, 1000);
    if (loading) {
           return <Load />
         }
   
  return (
     
    <div className="app" >
     <header><p className="title">To Watch </p></header>
     
      <Switch>
                
                <Route path="/" exact component={H} />
                <PrivateRoute  path="/home" exact component={Home} />
                <PrivateRoute path="/account"exact  component={Account} />
                <PrivateRoute path="/updatePseudo" exact component={Pseudo} />
                <PrivateRoute path="/updateMail" component={Mail} />
                <PrivateRoute path="/updatePassword" exact component={Password} />
                <PrivateRoute path="/media/:media/:id" exact  component={Media} />
                <PrivateRoute path="/genre/:id" exact component={Genre} />
                <PrivateRoute path="/genretv/:id" exact component={GenreTv} />
                <PrivateRoute path="/mentions" exact component={Mentions} />
                <PrivateRoute path="/forbidden" exact  component={Forbidden} />
                <PublicRoute path="/signup" exact  component={Signup} />
                <PublicRoute path="/login" exact  component={Login}/>
                <PrivateRoute path="/admin" exact component={Admin}/>
                <Route path="/404" component={NotFoundPage} />
                    <Redirect to="/404" />
            </Switch> 
    </div>
    
  );
}

export default App;
