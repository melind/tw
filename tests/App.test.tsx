import React from 'react';
import { shallow, mount, render} from 'enzyme';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../src/components/App';
/** test to check that we don't broke our code in the evolution */


/* snapshot tool let us to follow the evlution of our app,
 it compare the dom of our component save in file snap and the actual dom*/

/* test de rendu*/
describe('App', () => {
it('App should render correctly', () => {
  const tree = renderer
    .create(<App />).toJSON();//use render() of component
  expect(tree).toMatchSnapshot();
});
/* test de fonctionnalitÃ©*/
 it('renders App', () => {
     const wrapper = shallow(<App optInValue={true} authenticated={true}/>);

     expect(wrapper.length).toEqual(1);   
  }); 



});
