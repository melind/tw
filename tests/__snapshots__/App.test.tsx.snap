// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should render correctly 1`] = `
<div
  className="load"
>
  <img
    alt="gif d'un décompte"
    src="https://media.giphy.com/media/VxbP9tLeKzazm/giphy.gif"
  />
  <p>
    “Powered by Giphy”
  </p>
</div>
`;
